### ISSUE TITLE
[Bug] Wrong unicode in some musics and albums

### ISSUE THREAD

**Issue by asimovc** [zyrouge/symphony#274]
### Description

Some albums with special characters don't are in the correct unicode, it sometimes creates two duplicate albums and that nota occours in other music players like Vanilla Music Player

### Steps to Reproduce

1. Download album
2. Just open the app

### Expected Behavior

Album and music with right unicode

### Screenshots

![IMG_20230915_042832](https://github.com/zyrouge/symphony/assets/142914286/3f5849f6-14eb-4641-a3f7-059463131312)

![IMG_20230915_043038](https://github.com/zyrouge/symphony/assets/142914286/223de01b-874e-4f04-9364-401000e71b94)

### Device

-   Device: [Google Pixel 7a]
-   OS: [Android 13]
-   Version [2023.8.105]

### Checklist

-   [x] I have read the [Contribution Guidelines](https://github.com/zyrouge/symphony/wiki/Contributions-Guidelines#issues).

**Comment by alpha-tango-kilo** [comment]
I've had this issue with non-unicode naming too - just apostrophes in my case

<details><summary>Screenshot</summary>
<p>

![image](https://github.com/zyrouge/symphony/assets/12728900/6f2c452f-32bc-48d4-a72e-5878d43a2978)

</p>
</details> 


I tried removing them from the file name as a mitigation, but that didn't change anything unfortunately

Device: Pixel 4a
OS: GrapheneOS 13
Version: `2023.8.105`

**Comment by zyrouge** [comment]
> I've had this issue with non-unicode naming too - just apostrophes in my case
> Screenshot
> 
> I tried removing them from the file name as a mitigation, but that didn't change anything unfortunately
> 
> Device: Pixel 4a OS: GrapheneOS 13 Version: `2023.8.105`

Could you send that file? I'll check for possible solutions.

**Comment by alpha-tango-kilo** [comment]
Email sent from the email on my profile to yours 😃 Thanks for taking a look!

**Comment by zyrouge** [comment]
> Email sent from the email on my profile to yours 😃 Thanks for taking a look!

Seems like I did not received any mail. Could you possibly send it here or in Discord?

**Comment by zyrouge** [comment]
It seems like MediaStore seems to transform unsupported unicode characters wrongly.
![image](https://github.com/zyrouge/symphony/assets/50770114/941b2c21-8220-4eb5-8137-ccfaf3e969e2)

**Comment by asimovc** [comment]
@zyrouge why other foss apps can solve it?

**Comment by zyrouge** [comment]
> @zyrouge why other foss apps can solve it?

They don't solve it. They probably don't use `MediaStore`. For example, Vanilla player does all the metadata retrieval by accessing files directly.

**Comment by asimovc** [comment]
Then why not make this?

**Comment by zyrouge** [comment]
> Then why not make this?

Because it is not easy. MediaStore is a baked within Android and it is meant to be used for applications such as this.

**Comment by asimovc** [comment]
Then it can be fixed is just a hard work

**Comment by alpha-tango-kilo** [comment]
@zyrouge did you file an upstream bug report on this, or is it a known issues?

Unicode is a hell of a drug, but I'm amazed Android can't handle even ASCII characters properly. Shame

**Comment by zyrouge** [comment]
> @zyrouge did you file an upstream bug report on this, or is it a known issues?
> 
> Unicode is a hell of a drug, but I'm amazed Android can't handle even ASCII characters properly. Shame

Nope, I couldn't find any relevant issues in Android issue tracker.

**Comment by alpha-tango-kilo** [comment]
Out of interest, I tried my same song in [another app](https://github.com/chr56/Phonograph_Plus) which I know uses MediaStore, and can confirm it had the exact same incorrect/garbled name :(

**Comment by zyrouge** [comment]
> Out of interest, I tried my same song in [another app](https://github.com/chr56/Phonograph_Plus) which I know uses MediaStore, and can confirm it had the exact same incorrect/garbled name :(
> 
> This definitely should be reported upstream

Someone can try doing it but they've been taking ages on bugs related to MediaStore metadata.
